src/imitation/algorithms/pebble/entropy_reward.py

```diff
@@ -0,0 +1,125 @@
+"""Reward function for the PEBBLE training algorithm."""
+
+from enum import Enum, auto
+from typing import Dict, Optional, Tuple, Union
+
+import numpy as np
+import torch as th
+
+from imitation.policies.replay_buffer_wrapper import (
+    ReplayBufferRewardWrapper,
+    ReplayBufferView,
+    ReplayBufferAwareRewardFn,
+)
+from imitation.rewards.reward_function import RewardFn
+from imitation.util import util
+from imitation.util.networks import RunningNorm
+
+
+class PebbleRewardPhase(Enum):
+    """States representing different behaviors for PebbleStateEntropyReward."""
+
+    UNSUPERVISED_EXPLORATION = auto()  # Entropy based reward
+    POLICY_AND_REWARD_LEARNING = auto()  # Learned reward
+
+
+class PebbleStateEntropyReward(ReplayBufferAwareRewardFn):
```

Comment:
No intention here, this was a natural result of the refactoring out of the original buffer wrapper.

I agree separating the two would be nice, and it would be easier to test. Two things complicate it, though:
* when there are not enough observations, we would need to fall back to something, eg., zero vector (which is probably fine)
* it complicates pickling. Either the entropy part would be aware of state changes with ReplayBufferView and the setting logic would be duplicated in both, or unaware but then we may loose entropy_stats during pickling (it would be easier to explain in code)

I may take a look later if there is time.

Reply:
Let's leave it as is for now given it'd be a fairly major change, but please don't resolve this comment (I might ask someone else to look at making the change before merging, or merge but make an issue to track it).